viadot/sources/sap_rfc.py

```diff
@@ -84,6 +85,137 @@ def trim_where(where: str) -> Tuple[str, OrderedDictType[str, str]]:
+        If so, remove the last rows added, to fit the size of the previous columns.
+
+    Args:
+        row_index (int): Number of rows set it down in he first iteration with the SAP table.
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        chunk (int): The part number in which a number of SAP table columns have been split.
+        fields (List[str]): A list with the names of the columns in a chunk.
+
+    Returns:
+        Union[int, np.array, bool]: A tuple with the parameters "row_index", "data_raw", a new
+            boolean variable "start" to indicate when the for loop has to be restarted,
+            and "chunk" variable.
+    """
+    start = False
+    if row_index == 0:
+        row_index = data_raw.shape[0]
+        if row_index == 0:
+            logger.warning(
+                f"Empty output was generated for chunk {chunk} in columns {fields}."
+            )
+            start = True
+    elif data_raw.shape[0] != row_index:
+        data_raw = data_raw[:row_index]
+        logger.warning(
+            f"New rows were generated during the execution of the script. The table is truncated to the number of rows for the first chunk"
+        )
+
+    return row_index, data_raw, start
+
+
+def replace_separator_in_data(
+    data_raw: np.array,
+    no_sep_index: np.array,
+    record_key: str,
+    pos_sep_index: np.array,
+    sep: str,
+    replacement: str,
+) -> np.array:
+    """Fundtion to replace the extra separator in every row of the data_raw numpy array.
+
+    Args:
+        data_raw (np.array): Array with the data retrieve from SAP table.
+        no_sep_index (np.array): Array with indexes where are extra separators characters in rows.
+        record_key (str): Key word to extract the data from the numpy array "data_raw".
+        pos_sep_index (np.array): Array with indexes where are placed real separators.
+        sep (str): Which separator to use when querying SAP.
+        replacement (str): In case of sep is on a columns, set up a new character to replace
+            inside the string to avoid flow breakdowns.
+
+    Returns:
+        np.array: the same data_raw numpy array with the "replacement" separator instead.
+    """
+    for no_sep in no_sep_index:
+        logger.warning(
+            "A separator character was found and replaced inside a string text that could produce future errors:"
+        )
+        logger.warning("\n" + data_raw[no_sep][record_key])
+        split_array = np.array([*data_raw[no_sep][record_key]])
+        position = np.where(split_array == f"{sep}")[0]
+        index_sep_index = np.argwhere(np.in1d(position, pos_sep_index) == False)
+        index_sep_index = index_sep_index.reshape(
+            len(index_sep_index),
+        )
+        split_array[position[index_sep_index]] = replacement
+        data_raw[no_sep][record_key] = "".join(split_array)
+        logger.warning("\n" + data_raw[no_sep][record_key])
```

Comment:
That is for clarity when it would be printed on the screen, of course, it is not necessary but I would recommend that just to better read it on the screen., when the error raises.